plugin/dnstap/dnstapio/io.go

```diff
@@ -65,36 +77,59 @@ func (dio *dnstapIO) Dnstap(payload tap.Dnstap) {
 	}
 }
 
+func (dio *dnstapIO) closeConnection() {
```

Comment:
My question was more; why can't all this moved to Close()? Why are there 2 functions that "close"?

Reply:
Function closeConnection() is used on reconnect (before newConnection() call), Close() is used on shutdown - it just close queue and this cause closeConnection() and exit from serve().